### DISCUSSION TITLE
Does blitzjs generate related types when using use query and an include input?

### DISCUSSION THREAD

**Discussion by Gim3l** [blitz-js/blitz#1598] [Q&A] [answered]
I haven't used blitz for a few week and I'm not sure if something has changed regarding this. Let's say I have a query.
```
import { Ctx } from "blitz"
import db, { Prisma } from "db"

type GetProjectType = {
  where: Prisma.FindFirstProjectArgs["where"]
  include?: Prisma.FindFirstProjectArgs["include"]
}

export default async function getProject({ gameId, offerType }: GetProjectType , ctx: Ctx) {
  ctx.session.authorize()
  return await db.project.findFirst({where, include})
}
```

Would I be able to call ```project.author``` in the frontend assuming I have author in the 'include' object when calling useQuery?
This does not work for me and only works if my query was ```await db.project.findFirst({where, {include: {author: true} }})```
Should I open an issue instead? Does anyone else experience this? Or, is this expected?

**Comment by flybayer** [comment, marked as answer]
Unfortunately that's a Typescript limitation currently. Hopefully will be fixed at some point.

So what I do instead is hardcode the include statement inside the query resolver.
